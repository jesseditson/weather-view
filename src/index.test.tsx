import React from 'react'
import renderer, { act } from 'react-test-renderer'
import Weather from './index'

test('Basic render works', () => {
    const component = renderer.create(
        <Weather apiKey="fake" location="San Francisco" />
    )
    expect(component.toJSON()).toMatchSnapshot()
})

test('It displays data returned from onGetWeatherData', () => {
    // TODO: this doesn't work yet
    return act(async () => {
        const getWeatherData = jest.fn().mockResolvedValue({
            location: 'Test',
            weatherName: 'bad',
            weatherDesc: 'rainy and bad',
            icon: 'http://openweathermap.org/img/wn/11d@2x.png',
            temp: 42,
            feelsLike: 42,
            tempMin: 42,
            tempMax: 42,
            pressure: 42,
            humidity: 42,
            sunrise: 0,
            sunset: 0,
        })
        const component = renderer.create(
            <Weather
                apiKey="fake"
                location="San Francisco"
                onGetWeatherData={getWeatherData}
            />
        )
        expect(getWeatherData).toHaveBeenCalledWith({
            apiKey: 'fake',
            location: 'San Francisco',
        })
        expect(component.toJSON()).toMatchSnapshot()
    })
})
