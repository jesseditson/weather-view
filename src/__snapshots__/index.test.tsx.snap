// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic render works 1`] = `
<div
  className="css-73phlt"
>
  Loading...
</div>
`;
